.github/CONTRIBUTING.md

```diff
@@ -1,33 +1,167 @@
+  [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and
+  [this tool](https://github.com/colinkeenan/silentcast) or
+  [this tool](https://github.com/GNOME/byzanz) on Linux.
 
 ## Branching Model
 
-(tbd)
+### Main Branches
+
+The central repo holds two main branches with an infinite lifetime:
+
+- **master**
+- **develop**
+
+We consider origin/master to be the main branch where the source code of HEAD always reflects a
+production-ready state, and origin/develop to be the main branch where the source code of HEAD
+always reflects a state with the latest delivered development changes for the next release.
+When the source code in the develop branch reaches a stable point and is ready to be released, all
+of the changes should be merged back into master. Therefore, each time when changes are merged back
+into master, this is a new production release by definition.
+
+### Supporting Branches
+
+Next to the main branches master and develop, our development model uses a few of supporting
+branches to aid parallel development between team members, ease tracking of features and to assist
+in quickly fixing live production problems. Unlike the main branches, these branches always have a
+limited life time, since they will be removed eventually.
+
+The two main different types of branches we may use are:
+
+- **Feature branches**
+- **Hotfix branches**
+
+**Creating a feature branch**
+
+- branch **from** -> `develop` -> must merge back **into** -> `develop`
+
+Naming convention:
+
+- anything except `master`, `develop`, `release*`, or `hotfix*`
+- preferred `feature/{issueId}-descriptive_feature_title`
+
+**Creating a hotfix branch**
+
+- branch **from** -> `master` -> must merge back **into** -> `master` and `develop`
+
+Naming convention:
+
+- `hotfix/{issueId}-descriptive_issue_title`
+- `hotfix/{new_patch_version}`
+
+Hotfix branches are created from the master branch. For example, say version 1.2 is the current
+production release running live and causing troubles due to a severe bug. But changes on develop
+are yet unstable. We may then branch off a hotfix branch and start fixing the problem:
+
+```sh
+git checkout -b hotfix/1.2.1 master
+npm version patch -m "Upgrade to %s for reasons"
+git commit -a -m "Bumped version number to 1.2.1"
+```
+
+Don’t forget to bump the version number after branching off!
+Then, fix the bug and commit the fix in one or more separate commits.
+
+```sh
+git commit -m "Fixed severe production problem"
+```
+
+## Pull Request Process
+
+Following is a short guide on how to make a valid Pull Request.
+
+1. Firstly you need a local fork of the the project, so go ahead and press the `fork` button in
+   GitHub. This will create a copy of the repository in your own GitHub account and you’ll see a
+   note that it’s been forked underneath the project name: `Forked from ZEISS/precise-ui`.
+   Clone the newly forked repo locally and set up a new remote that points to the original project
```

Comment:
"repository" or "repo" (as before I personally would always call it "repository").